### PR TITLE
helm: Discord access-request helm chart

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR introduces the Discord access-request plugin Helm chart. Part of https://github.com/gravitational/teleport/issues/9563

~Blocked by https://github.com/gravitational/teleport-plugins/pull/667~

### CODE REVIEW

File: charts/access/discord/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: "10.3.2"
```
Comment (marcoandredinis) [inline charts/access/discord/Chart.yaml:18]:
```suggestion
version: "11.0.1"
```

File: charts/access/discord/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "10.3.2"
```
Comment (marcoandredinis) [inline charts/access/discord/Chart.yaml:24]:
```suggestion
version: "11.0.1"
```

File: charts/access/discord/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Teleport Access Request Discord Plugin
+
+This chart sets up and configures a Deployment for the Access Request Discord plugin.
+
+## Installation
+
+See the [Access Requests with Discord guide](https://goteleport.com/docs/access-controls/access-request-plugins/ssh-approval-discord/)
```
Comment (marcoandredinis) [inline charts/access/discord/README.md:7]: This link returns 404
I assume we'll a PR with proper instructions later on

File: charts/access/discord/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    <td>The path of the token in the secret described by <code>discord.tokenFromSecret</code></td>
+    <td>string</td>
+    <td><code>"discordToken"</code></td>
+    <td>no</td>
+  </tr>
+
+  <tr>
+    <td><code>roleToRecipients</code></td>
+    <td>
+      Mapping of roles to a list of channels and Discord emails. <br />
```
Comment (marcoandredinis) [inline charts/access/discord/README.md:70]: We can't really use Discord emails right? Only IDs (or am I mixing this with MS Teams?)

File: charts/access/discord/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - /usr/local/bin/teleport-plugin
```
Comment (marcoandredinis) [inline charts/access/discord/templates/deployment.yaml:34]: Should it be `teleport-plugin-discord`?

**Review by marcoandredinis — COMMENTED** [review]
This looks mostly good :+1: 

I'm ignoring everythin that belongs to `access/`
That should be from other PR, right?

Versions in `.snap` files mismatch our current release

File: charts/access/discord/README.md (hunk 3 above)
Comment (hugoShaka) [reply charts/access/discord/README.md:7]: 👍 https://github.com/gravitational/teleport/pull/17593

File: charts/access/discord/templates/deployment.yaml (hunk 5 above)
Comment (hugoShaka) [reply charts/access/discord/templates/deployment.yaml:34]: I'm not sure why, maybe we wanted to standardize everything in the same helm chart or docker image, but all other plugins are named `teleport-plugin` 🤷

File: charts/access/discord/README.md (hunk 4 above)
Comment (hugoShaka) [reply charts/access/discord/README.md]: fixed

File: charts/access/discord/values.schema.json [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+            "default": {},
+            "examples": [
+                {
+                    "dev": [
+                        "devs-discord-channel"
+                    ],
+                    "*": [
+                        "admin@email.com",
+                        "admin-discord-channel"
+                    ]
```
Comment (marcoandredinis) [inline charts/access/discord/values.schema.json:340]: :nail_care: 
I would try to keep the examples as close as possible to real values

**Review by marcoandredinis — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
Bot.

**Pull request merged**